**<파일 경로>**
- C:/Users/Brian/Desktop/Data Science/Projects/Playdata/dataset/subway

In [1]:
import pandas as pd
from tqdm.notebook import tqdm
from functools import reduce

In [2]:
filepath = input()

C:/Users/Brian/Desktop/Data Science/Projects/Playdata/dataset/subway


# 2018~2019년도 지하철 승하차 인원 데이터 셋 결합
- 연도별 서울특별시 지하철 승하차 인원 데이터 셋을 결합해주겠다.

In [3]:
subway_boarding_2018 = pd.read_excel('{}/서울교통공사 2018년 일별 역별 시간대별 승하차인원(1_8호선).xlsx'.format(filepath), header = 1)
subway_boarding_2019 = pd.read_excel('{}/서울교통공사 2019년 일별 역별 시간대별 승하차인원(1_8호선).xlsx'.format(filepath), header = 1)

In [4]:
subway_boarding_2018.head()

날짜   호선  역번호   역명  구분  05 ~ 06  06 ~ 07  07 ~ 08  08 ~ 09  09 ~ 10  \
0 2018-01-01  1호선  150  서울역  승차      373      318      365      785     1047   
1 2018-01-01  1호선  150  서울역  하차      205     1040      872      984     1650   
2 2018-01-01  1호선  151   시청  승차       87      105      124      197      291   
3 2018-01-01  1호선  151   시청  하차       47      294      497     1017      673   
4 2018-01-01  1호선  152   종각  승차      604      399      191      250      370   

   ...  16 ~ 17  17 ~ 18  18 ~ 19  19 ~ 20  20 ~ 21  21 ~ 22  22 ~ 23  \
0  ...     3476     3891     3227     2945     2382     3070     1750   
1  ...     2885     2845     2337     2131     1669     1404      868   
2  ...     1027     1102     1278     1163     1032      975      553   
3  ...      781      702      552      388      308      236      160   
4  ...     2127     2172     2171     1873     1935     2084     1458   

   23 ~ 24  00 ~ 01    합 계  
0      781       96  40393  
1      477      147  34730  
2      214        9  12298  
3      100       39  10945  
4      580       28  22621  

[5 rows x 26 columns]

In [5]:
subway_boarding_2019.tail()

날짜   호선   역번호  역명  구분  05 ~ 06  06 ~ 07  07 ~ 08  08 ~ 09  \
200799 2019-12-31  8호선  2825  신흥  하차       20       87       86      243   
200800 2019-12-31  8호선  2826  수진  승차      111      224      542      813   
200801 2019-12-31  8호선  2826  수진  하차       27       97      115      433   
200802 2019-12-31  8호선  2827  모란  승차       73       95      342      482   
200803 2019-12-31  8호선  2827  모란  하차       15       99      140      386   

        09 ~ 10  ...  16 ~ 17  17 ~ 18  18 ~ 19  19 ~ 20  20 ~ 21  21 ~ 22  \
200799      208  ...      501      510      562      493      378      301   
200800      436  ...      427      436      395      245      219      155   
200801      259  ...      373      463      501      452      354      332   
200802      253  ...      366      375      418      242      199      152   
200803      189  ...      306      277      403      279      213      195   

        22 ~ 23  23 ~ 24  00 ~ 01   합 계  
200799      257      154      163  5794  
200800      126       68       82  6228  
200801      322      256      163  5996  
200802      189      126       65  5151  
200803      186      130      407  4436  

[5 rows x 26 columns]

- 이제 2018~2019년도 지하철 승하차 인원 데이터 셋을 하나로 결합해주겠다.

In [6]:
subway_boarding_total = pd.concat([subway_boarding_2018, subway_boarding_2019], axis = 0)
subway_boarding_total

날짜   호선   역번호   역명  구분  05 ~ 06  06 ~ 07  07 ~ 08  08 ~ 09  \
0      2018-01-01  1호선   150  서울역  승차      373      318      365      785   
1      2018-01-01  1호선   150  서울역  하차      205     1040      872      984   
2      2018-01-01  1호선   151   시청  승차       87      105      124      197   
3      2018-01-01  1호선   151   시청  하차       47      294      497     1017   
4      2018-01-01  1호선   152   종각  승차      604      399      191      250   
...           ...  ...   ...  ...  ..      ...      ...      ...      ...   
200799 2019-12-31  8호선  2825   신흥  하차       20       87       86      243   
200800 2019-12-31  8호선  2826   수진  승차      111      224      542      813   
200801 2019-12-31  8호선  2826   수진  하차       27       97      115      433   
200802 2019-12-31  8호선  2827   모란  승차       73       95      342      482   
200803 2019-12-31  8호선  2827   모란  하차       15       99      140      386   

        09 ~ 10  ...  16 ~ 17  17 ~ 18  18 ~ 19  19 ~ 20  20 ~ 21  21 ~ 22  \
0          1047  ...     3476     3891     3227     2945     2382     3070   
1          1650  ...     2885     2845     2337     2131     1669     1404   
2           291  ...     1027     1102     1278     1163     1032      975   
3           673  ...      781      702      552      388      308      236   
4           370  ...     2127     2172     2171     1873     1935     2084   
...         ...  ...      ...      ...      ...      ...      ...      ...   
200799      208  ...      501      510      562      493      378      301   
200800      436  ...      427      436      395      245      219      155   
200801      259  ...      373      463      501      452      354      332   
200802      253  ...      366      375      418      242      199      152   
200803      189  ...      306      277      403      279      213      195   

        22 ~ 23  23 ~ 24  00 ~ 01    합 계  
0          1750      781       96  40393  
1           868      477      147  34730  
2           553      214        9  12298  
3           160      100       39  10945  
4          1458      580       28  22621  
...         ...      ...      ...    ...  
200799      257      154      163   5794  
200800      126       68       82   6228  
200801      322      256      163   5996  
200802      189      126       65   5151  
200803      186      130      407   4436  

[401554 rows x 26 columns]

- 다음으로 변수명을 다음과 같이 변경해주겠다.
    - 날짜 = date
    - 호선 = line
    - 역번호 = station_num
    - 역명 = station_name
    - 구분 = boarding
    - 05 ~ 06 = 5
    - 06 ~ 07 = 6
    - 07 ~ 08 = 7
    - 08 ~ 09 = 8
    - 09 ~ 10 = 9
    - 10 ~ 11 = 10
    - 11 ~ 12 = 11
    - 12 ~ 13 = 12
    - 13 ~ 14 = 13
    - 14 ~ 15 = 14
    - 15 ~ 16 = 15
    - 16 ~ 17 = 16
    - 17 ~ 18 = 17
    - 18 ~ 19 = 18
    - 19 ~ 20 = 19
    - 20 ~ 21 = 20
    - 21 ~ 22 = 21
    - 22 ~ 23 = 22
    - 23 ~ 24 = 23
    - 00 ~ 01 = 0
    - 합 계 = total
- 그리고 '합 계' 변수는 분석에 필요하지 않으므로 제거하도록 하겠다.

In [7]:
# 변수명 변경
subway_boarding_total.rename(columns = {'날짜': 'date', '호선': 'line', '역번호': 'station_num', '역명': 'station_name', '구분': 'boarding',
                                        '05 ~ 06': 5, '06 ~ 07': 6, '07 ~ 08': 7, '08 ~ 09': 8,
                                        '09 ~ 10': 9, '10 ~ 11': 10, '11 ~ 12': 11, '12 ~ 13': 12,
                                        '13 ~ 14': 13, '14 ~ 15': 14, '15 ~ 16': 15, '16 ~ 17': 16,
                                        '17 ~ 18': 17, '18 ~ 19': 18, '19 ~ 20': 19, '20 ~ 21': 20,
                                        '21 ~ 22': 21, '22 ~ 23': 22, '23 ~ 24': 23, '00 ~ 01': 0,
                                        '합 계': 'total'}, inplace = True)
# '합 계(total)' 변수 제거
subway_boarding_total.drop(['total'], axis = 1, inplace = True)

In [8]:
# date 변수의 타입을 datetime으로 변경
subway_boarding_total['date'] = pd.to_datetime(subway_boarding_total['date'])
# date를 기준으로 오름차순 정렬
subway_boarding_total.sort_values(by = 'date', inplace = True)

In [9]:
subway_boarding_total

date line  station_num station_name boarding    5    6     7  \
0      2018-01-01  1호선          150          서울역       승차  373  318   365   
7590   2018-01-01  7호선         2729         건대입구       승차  355  313   181   
7589   2018-01-01  7호선         2728  어린이대공원(세종대)       하차   34  141   127   
7588   2018-01-01  7호선         2728  어린이대공원(세종대)       승차  147  177   178   
7587   2018-01-01  7호선         2727       군자(능동)       하차   13  131    86   
...           ...  ...          ...          ...      ...  ...  ...   ...   
200436 2019-12-31  3호선          341         경찰병원       승차   69  218   802   
200437 2019-12-31  3호선          341         경찰병원       하차   13  108   253   
200438 2019-12-31  3호선          342           오금       승차  120  243   922   
200432 2019-12-31  3호선          339           수서       승차  170  438  1285   
200803 2019-12-31  8호선         2827           모란       하차   15   99   140   

           8     9  ...    15    16    17    18    19    20    21    22   23  \
0        785  1047  ...  2980  3476  3891  3227  2945  2382  3070  1750  781   
7590     143   196  ...   762   793   795   873   714   719   766   661  371   
7589     186   261  ...   494   473   441   447   378   395   412   334  242   
7588     201   254  ...   508   566   604   447   326   295   339   314  138   
7587      98   142  ...   337   371   415   387   343   313   313   320  195   
...      ...   ...  ...   ...   ...   ...   ...   ...   ...   ...   ...  ...   
200436   998   542  ...   479   562   668   653   293   233   191   166   74   
200437  1201   580  ...   413   465   600   678   532   334   280   308  159   
200438  1182   534  ...   336   331   387   422   198   145   138   148   68   
200432  1621  1007  ...  1261  1412  1470  1708   990   827   605   586  297   
200803   386   189  ...   235   306   277   403   279   213   195   186  130   

          0  
0        96  
7590      4  
7589     14  
7588      1  
7587     14  
...     ...  
200436   17  
200437  110  
200438   19  
200432  112  
200803  407  

[401554 rows x 25 columns]

## 데이터 셋의 형태 변경
- 현재 주어진 데이터 셋의 형태를 보면, 시간대별 승/하차 인원 변수(0, 5 ~ 23)가 컬럼으로 들어가있다.
- 때문에 시간대별 승/하차 인원 변수(0, 5 ~ 23)를 시간대(hour) 변수와 승/하차 인원수(boarding_num) 변수로 분리되도록 데이터 셋의 형태를 변환해주도록 하겠다.
    - 즉, **columns to rows** 변환

In [10]:
# '0, 5~23' 변수들이 문자열로 입력되어 있다는 점을 고려해서 리스트에 담아주기
pivot_values = []
for i in list(range(5, 24)) + [0]:
    pivot_values.append(i)
    
# 리스트에 담긴 숫자들을 오름차순 정렬
pivot_values = sorted(pivot_values)
pivot_values

[0, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23]

In [11]:
# DataFrame to Pivot Table
subway_pivot = pd.pivot_table(subway_boarding_total, index = ['date', 'line', 'station_num', 'station_name', 'boarding'], 
                              values = pivot_values)
subway_pivot

0    5     6    7     8   \
date       line station_num station_name boarding                              
2018-01-01 1호선  150         서울역          승차         96  373   318  365   785   
                                         하차        147  205  1040  872   984   
                151         시청           승차          9   87   105  124   197   
                                         하차         39   47   294  497  1017   
                152         종각           승차         28  604   399  191   250   
...                                                ...  ...   ...  ...   ...   
2019-12-31 8호선  2825        신흥           하차        163   20    87   86   243   
                2826        수진           승차         82  111   224  542   813   
                                         하차        163   27    97  115   433   
                2827        모란           승차         65   73    95  342   482   
                                         하차        407   15    99  140   386   

                                                     9     10    11    12  \
date       line station_num station_name boarding                           
2018-01-01 1호선  150         서울역          승차        1047  1576  2510  3233   
                                         하차        1650  1743  2175  2991   
                151         시청           승차         291   499   722   612   
                                         하차         673   657   820   958   
                152         종각           승차         370   439   705   980   
...                                                 ...   ...   ...   ...   
2019-12-31 8호선  2825        신흥           하차         208   189   205   327   
                2826        수진           승차         436   343   267   350   
                                         하차         259   198   207   283   
                2827        모란           승차         253   210   253   281   
                                         하차         189   175   215   142   

                                                     13    14    15    16  \
date       line station_num station_name boarding                           
2018-01-01 1호선  150         서울역          승차        3145  2443  2980  3476   
                                         하차        2877  2743  2687  2885   
                151         시청           승차         580   821   907  1027   
                                         하차        1009   877   830   781   
                152         종각           승차        1153  1392  1710  2127   
...                                                 ...   ...   ...   ...   
2019-12-31 8호선  2825        신흥           하차         347   381   382   501   
                2826        수진           승차         297   285   407   427   
                                         하차         381   376   404   373   
                2827        모란           승차         377   350   303   366   
                                         하차         224   220   235   306   

                                                     17    18    19    20  \
date       line station_num station_name boarding                           
2018-01-01 1호선  150         서울역          승차        3891  3227  2945  2382   
                                         하차        2845  2337  2131  1669   
                151         시청           승차        1102  1278  1163  1032   
                                         하차         702   552   388   308   
                152         종각           승차        2172  2171  1873  1935   
...                                                 ...   ...   ...   ...   
2019-12-31 8호선  2825        신흥           하차         510   562   493   378   
                2826        수진           승차         436   395   245   219   
                                         하차         463   501   452   354   
                2827        모란           승차         375   418   242   199   
                                         하차         277   403   279   213   

     

In [12]:
# stack 함수를 통한 변환 후, DataFrame으로 변경
subway_boarding_total2 = pd.DataFrame(subway_pivot.stack()).reset_index()
subway_boarding_total2

date line  station_num station_name boarding  level_5    0
0       2018-01-01  1호선          150          서울역       승차        0   96
1       2018-01-01  1호선          150          서울역       승차        5  373
2       2018-01-01  1호선          150          서울역       승차        6  318
3       2018-01-01  1호선          150          서울역       승차        7  365
4       2018-01-01  1호선          150          서울역       승차        8  785
...            ...  ...          ...          ...      ...      ...  ...
8031075 2019-12-31  8호선         2827           모란       하차       19  279
8031076 2019-12-31  8호선         2827           모란       하차       20  213
8031077 2019-12-31  8호선         2827           모란       하차       21  195
8031078 2019-12-31  8호선         2827           모란       하차       22  186
8031079 2019-12-31  8호선         2827           모란       하차       23  130

[8031080 rows x 7 columns]

- stack 함수를 통해 만들어진 DataFrame의 level_5 변수와 0 변수의 변수명을 변경해주겠다.
    - level_5 = hour
    - 0 = boarding_num
- 그리고 만들어진 데이터 셋을 date와 hour 변수를 기준으로 오름차순 정렬해주도록 하겠다.

In [13]:
subway_boarding_total2.rename(columns = {'level_5': 'hour', 0: 'boarding_num'}, inplace = True)
subway_boarding_total2.sort_values(by = ['date', 'hour'], axis = 0, inplace = True)
subway_boarding_total2

date line  station_num station_name boarding  hour  boarding_num
0       2018-01-01  1호선          150          서울역       승차     0            96
20      2018-01-01  1호선          150          서울역       하차     0           147
40      2018-01-01  1호선          151           시청       승차     0             9
60      2018-01-01  1호선          151           시청       하차     0            39
80      2018-01-01  1호선          152           종각       승차     0            28
...            ...  ...          ...          ...      ...   ...           ...
8030999 2019-12-31  8호선         2825           신흥       하차    23           154
8031019 2019-12-31  8호선         2826           수진       승차    23            68
8031039 2019-12-31  8호선         2826           수진       하차    23           256
8031059 2019-12-31  8호선         2827           모란       승차    23           126
8031079 2019-12-31  8호선         2827           모란       하차    23           130

[8031080 rows x 7 columns]

# 지하철 역 정보 데이터 셋 결합
- 아래 4개의 데이터 셋을 하나로 결합해주겠다.
    1. 지하철 역사 면적(대합실, 승강장) 데이터 셋
    2. 지하철 역 주소 데이터 셋
    3. 지하철 역 형식(섬식, 상대식, 복합식) 데이터 셋
    4. 지하철 역 환승노선/출입구 데이터 셋

**<참고>**
- 각 데이터 셋에 존재하는 지하철 역명이 서로 달랐기 때문에, Excel을 사용하여 '지하철 승하차 인원 데이터 셋'에 존재하는 'station_name(역명)'과 동일하게 만들어주었다.
    - 이렇게 해야 추후에 지하철 승하차 인원 데이터 셋과 결합 시 문제가 발생하지 않는다.
- Excel을 통해 한글로 된 변수명들을 모두 영어로 변경해주었고, 불필요한 변수들은 바로 제거해주었다.
- 그리고 지하철 역 주소 데이터 셋의 'gu(구)' 변수 중, "강남"으로 되어있는 것들을 모두 "강남구"라고 변경해주었다.
    - 추후에 기상 데이터와 결합 시, "강남"이라고 되어있으면 문제가 발생하기 때문이다.
    - ex) "강남" >> "강남구"
- 또한 지하철 역 주소 데이터 셋을 확인하던 중, 'address(상세주소)' 변수에서 "서울**특**별시"인데 "서울**툭**별시"로 잘못 기입된 부분이 있었다.
    - 따라서 Excel에서 해당 부분을 수정해주었다.
    - ex) "서울**툭**별시" >> "서울**특**별시"

In [14]:
subway_area = pd.read_csv("{}/서울교통공사_역사 면적 정보.csv".format(filepath))
subway_address = pd.read_excel("{}/서울교통공사_역별 주소 현황.xlsx".format(filepath))
subway_type = pd.read_excel('{}/서울교통공사_역별 형식 정보.xlsx'.format(filepath))
subway_info = pd.read_csv('{}/서울교통공사_역별 환승노선&출입구 정보.csv'.format(filepath))

## 9호선 제거
- 분석에는 1~8호선까지의 데이터만을 사용할 것이므로, 9호선에 대한 정보는 모두 제거해주겠다.

In [15]:
subway_area = subway_area[subway_area['line'] != 9]
subway_address = subway_address[subway_address['line'] != 9]
subway_type = subway_type[subway_type['line'] != 9]
subway_info = subway_info[subway_info['line'] != 9]

In [16]:
subway_area

line station_name  waiting_area  platform_area
0       3         가락시장      12557.98        3508.00
1       8         가락시장       5110.63        2617.50
2       7      가산디지털단지      10116.83        2434.70
3       2           강남       4018.00        2374.00
4       7         강남구청       7987.99        2371.32
..    ...          ...           ...            ...
287     3           홍제       7220.00        1876.00
288     5           화곡       7144.92        1982.74
289     6  화랑대(서울여대입구)       7497.20        2610.02
290     4    회현(남대문시장)       5253.00        2702.00
291     6        효창공원앞       4771.62        2022.17

[279 rows x 4 columns]

In [17]:
subway_address

station_name  line    gu dong                             address
0           가락시장     3   송파구  가락동          서울특별시 송파구 가락동 184-23 가락시장역
1           가락시장     8   송파구  가락동  서울특별시 송파구 가락동 184-23 가락시장역(3, 8호선)
2        가산디지털단지     7   금천구  가산동         서울특별시 금천구 가산동 54-3 가산디지털단지역
3             강남     2   강남구  역삼동               서울특별시 강남구 역삼동 858 강남역
4           강남구청     7   강남구  삼성동          서울특별시 강남구 삼성동 111-44 강남구청역
..           ...   ...   ...  ...                                 ...
274           홍제     3  서대문구  홍제동            서울특별시 서대문구 홍제동 161-1 홍제역
275           화곡     5   강서구  화곡동           서울특별시 강서구 화곡동 1089-54 화곡역
276  화랑대(서울여대입구)     6   노원구  공릉동            서울특별시 노원구 공릉동 285-2 화랑대역
277    회현(남대문시장)     4    중구  남창동           서울특별시 중구 남창동 64-1 회현역 4호선
278        효창공원앞     6   용산구  효창동             서울특별시 용산구 효창동 80 효창공원앞역

[279 rows x 5 columns]

In [18]:
subway_type

line station_name type  length floor
0       3         가락시장  상대식     205    B4
1       8         가락시장  상대식     125    B2
2       7      가산디지털단지  상대식     165    B4
3       2           강남  상대식     205    B2
4       7         강남구청  상대식     165    B3
..    ...          ...  ...     ...   ...
286     3           홍제   섬식     205    B3
287     5           화곡   섬식     165    B2
288     6  화랑대(서울여대입구)  상대식     165    B2
289     4    회현(남대문시장)   섬식     205    B4
290     6        효창공원앞   섬식     165    B3

[278 rows x 5 columns]

In [19]:
subway_info

line      station_name   transfer  transfer_num  entrance  \
0       1               서울역  1,4,공항,경의             4         9   
1       1                시청        1,2             2         7   
2       1                종각        NaN             0         7   
3       1              종로3가      1,3,5             3         8   
4       1              종로5가        NaN             0         8   
..    ...               ...        ...           ...       ...   
272     8  남한산성입구(성남법원.검찰청)        NaN             0         4   
273     8             단대오거리        NaN             0         7   
274     8                신흥        NaN             0         4   
275     8                수진        NaN             0         4   
276     8                모란       8,분당             2         4   

                              gate  
0       1, 2, 3, 4, 5, 6, 7, 8, 9,  
1             1, 2, 3, 4, 5, 6, 7,  
2           1, 2, 3, 3-1, 4, 5, 6,  
3    1, 2, 10, 11, 12, 13, 14, 15,  
4          1, 2, 3, 4, 5, 6, 7, 8,  
..                             ...  
272                        1,2,3,4  
273                  1,2,3,4,5,6,7  
274                        1,2,3,4  
275                        1,2,3,4  
276                     9,10,11,12  

[277 rows x 6 columns]

## station_name(역명) 통일
- 기준 데이터 셋인 subway_boarding_total2(2018~2019년 지하철 승하차 인원 데이터 셋)에 존재하는 station_name으로 모든 데이터 셋의 station_name 변수를 통일시켜주겠다.

- 2018~2019년 지하철 승하차 인원 데이터 셋에 존재하는 (고유한) 지하철 역의 개수는 총 242개이다.

In [20]:
len(set(subway_boarding_total2['station_name']))

242

- 아래 4개의 데이터 셋 각각에 존재하는 (고유한) 지하철 역의 개수는 다음과 같다.
    1. 지하철 역사 면적(대합실, 승강장) 데이터 셋
        - 총 243개
    2. 지하철 역 주소 데이터 셋
        - 총 243개
    3. 지하철 역 형식(섬식, 상대식, 복합식) 데이터 셋
        - 총 243개
    4. 지하철 역 환승노선/출입구 데이터 셋
        - 총 242개

In [21]:
len(set(subway_area['station_name'])), len(set(subway_address['station_name'])), \
len(set(subway_type['station_name'])), len(set(subway_info['station_name']))

(243, 243, 243, 242)

- 2018~2019년 지하철 승하차 인원 데이터 셋에 존재하는 (고유한) 지하철 역의 개수는 총 242개인데, 위 4개의 데이터 셋에 존재하는 (고유한) 지하철 역의 개수와 불일치한다.
    - 어떤 지하철 역 때문에 불일치하는지 확인해보겠다.
    - 그래야 데이터 결합 시 문제가 발생하지 않는다!

In [22]:
# 2018~2019 지하철 승하차 인원 데이터 셋에 존재하는 (고유한) 지하철 역 개수
base_station_name = list(set(subway_boarding_total2['station_name']))
# 지하철 역사 면적(대합실, 승강장) 데이터 셋에 존재하는 (고유한) 지하철 역 개수
area_station_name = list(set(subway_area['station_name']))
# 지하철 역 주소 데이터 셋에 존재하는 (고유한) 지하철 역 개수
address_station_name = list(set(subway_address['station_name']))
# 지하철 역 형식(섬식, 상대식, 복합식) 데이터 셋에 존재하는 (고유한) 지하철 역 개수
type_station_name = list(set(subway_type['station_name']))
# 지하철 역 환승노선/출입구 데이터 셋에 존재하는 (고유한) 지하철 역 개수
info_station_name = list(set(subway_info['station_name']))

area_diff = []
for station_name in area_station_name:
    # base_station_name에 존재하지 않는 station_name을 추출
    if station_name not in base_station_name:
        area_diff.append(station_name)
        
address_diff = []
for station_name in address_station_name:
    # base_station_name에 존재하지 않는 station_name을 추출
    if station_name not in base_station_name:
        address_diff.append(station_name)
        
type_diff = []
for station_name in type_station_name:
    # base_station_name에 존재하지 않는 station_name을 추출
    if station_name not in base_station_name:
        type_diff.append(station_name)
        
info_diff = []
for station_name in info_station_name:
    # base_station_name에 존재하지 않는 station_name을 추출
    if station_name not in base_station_name:
        info_diff.append(station_name)

1. 지하철 역사 면적(대합실, 승강장) 데이터 셋에는 존재하지만, 2018~2019년 지하철 승하차 인원 데이터 셋에는 존재하지 않는 지하철 역들은 다음과 같다.
    - 신내
2. 지하철 역 주소 데이터 셋에는 존재하지만, 2018~2019년 지하철 승하차 인원 데이터 셋에는 존재하지 않는 지하철 역들은 다음과 같다.
    - 신내
3. 지하철 역 형식(섬식, 상대식, 복합식) 데이터 셋에는 존재하지만, 2018~2019년 지하철 승하차 인원 데이터 셋에는 존재하지 않는 지하철 역들은 다음과 같다.
    - 신내
4. 지하철 역 환승노선/출입구 데이터 셋에는 존재하지만, 2018~2019년 지하철 승하차 인원 데이터 셋에는 존재하지 않는 지하철 역들은 다음과 같다.
    - 지하철 역명이 완전히 일치한다 (즉, 수정할 필요가 없음)

In [23]:
area_diff

['신내']

In [24]:
address_diff

['신내']

In [25]:
type_diff

['신내']

In [26]:
info_diff

[]

- 확인해보니 '신내'역 때문에 지하철 역명이 불일치했음을 알 수 있었다.
- 따라서 2018~2019년 지하철 승하차 인원 데이터 셋에 존재하지 않는 '신내'역은 제거해주도록 하겠다.

In [27]:
subway_area = subway_area[subway_area['station_name'] != '신내']
subway_address = subway_address[subway_address['station_name'] != '신내']
subway_type = subway_type[subway_type['station_name'] != '신내']

In [28]:
# 각 데이터 셋에 존재하는 (고유한) 지하철 역의 개수 재확인
len(set(subway_area['station_name'])), len(set(subway_address['station_name'])), \
len(set(subway_type['station_name'])), len(set(subway_info['station_name']))

(242, 242, 242, 242)

## 4개의 데이터 셋 최종 결합
1. 지하철 역사 면적(대합실, 승강장) 데이터 셋
2. 지하철 역 주소 데이터 셋
3. 지하철 역 형식(섬식, 상대식, 복합식) 데이터 셋
4. 지하철 역 환승노선/출입구 데이터 셋

- 이제 위 4개의 데이터 셋을 'station_name'과 'line'을 조인키로 사용하여 inner join 해주겠다.

In [29]:
dfs = [subway_area, subway_address, subway_type, subway_info]
subway_information_total = reduce(lambda df_left, df_right: pd.merge(df_left, df_right, how = 'inner', on = ['station_name', 'line']), dfs)
subway_information_total

line station_name  waiting_area  platform_area    gu dong  \
0       3         가락시장      12557.98        3508.00   송파구  가락동   
1       8         가락시장       5110.63        2617.50   송파구  가락동   
2       7      가산디지털단지      10116.83        2434.70   금천구  가산동   
3       2           강남       4018.00        2374.00   강남구  역삼동   
4       7         강남구청       7987.99        2371.32   강남구  삼성동   
..    ...          ...           ...            ...   ...  ...   
272     3           홍제       7220.00        1876.00  서대문구  홍제동   
273     5           화곡       7144.92        1982.74   강서구  화곡동   
274     6  화랑대(서울여대입구)       7497.20        2610.02   노원구  공릉동   
275     4    회현(남대문시장)       5253.00        2702.00    중구  남창동   
276     6        효창공원앞       4771.62        2022.17   용산구  효창동   

                                address type  length floor transfer  \
0            서울특별시 송파구 가락동 184-23 가락시장역  상대식     205    B4      3,8   
1    서울특별시 송파구 가락동 184-23 가락시장역(3, 8호선)  상대식     125    B2      3,8   
2           서울특별시 금천구 가산동 54-3 가산디지털단지역  상대식     165    B4      1,7   
3                 서울특별시 강남구 역삼동 858 강남역  상대식     205    B2    2,신분당   
4            서울특별시 강남구 삼성동 111-44 강남구청역  상대식     165    B3     7,분당   
..                                  ...  ...     ...   ...      ...   
272            서울특별시 서대문구 홍제동 161-1 홍제역   섬식     205    B3      NaN   
273           서울특별시 강서구 화곡동 1089-54 화곡역   섬식     165    B2      NaN   
274            서울특별시 노원구 공릉동 285-2 화랑대역  상대식     165    B2      NaN   
275           서울특별시 중구 남창동 64-1 회현역 4호선   섬식     205    B4      NaN   
276             서울특별시 용산구 효창동 80 효창공원앞역   섬식     165    B3   6,경의중앙   

     transfer_num  entrance                        gate  
0               2         2                       7, 8,  
1               2         7             1,2,2-1,3,4,5,6  
2               2         5                   2,3,4,5,6  
3               2         8  1, 2, 7, 8, 9, 10, 11, 12,  
4               2         5                 1,2,3,3-1,4  
..            ...       ...                         ...  
272             0         4                1, 2, 3, 4,   
273             0         8             1,2,3,4,5,6,7,8  
274             0         7               1,2,3,4,5,6,7  
275             0         7               1,2,3,4,5,6,7  
276             2         4                     1,2,3,6  

[277 rows x 14 columns]

In [30]:
# 최종 결합된 데이터 셋에 존재하는 (고유한) 지하철 역의 개수 확인
len(set(subway_information_total['station_name']))

242

In [31]:
subway_information_total.isnull().sum()

line               0
station_name       0
waiting_area       0
platform_area      0
gu                 0
dong               0
address            0
type               0
length             0
floor              0
transfer         178
transfer_num       0
entrance           0
gate               4
dtype: int64

- transfer 변수와 gate 변수에는 NaN 값이 존재하는 것이 정상적이다.

- 끝으로 subway_boarding_total2 데이터 셋과 결합해주기 위해, 숫자로 되어 있는 'line' 변수 뒤에 '호선'이라는 글자를 붙여주겠다.
    - ex) "1" >> "1호선"

In [32]:
for i in range(len(subway_information_total)):
    subway_information_total['line'][i] = str(subway_information_total['line'][i]) + '호선'

subway_information_total.line.value_counts()

<ipython-input-32-caea5d198617>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  subway_information_total['line'][i] = str(subway_information_total['line'][i]) + '호선'
C:\Users\Brian\anaconda3\lib\site-packages\pandas\core\indexing.py:671: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_with_indexer(indexer, value)


7호선    51
5호선    51
2호선    50
6호선    38
3호선    34
4호선    26
8호선    17
1호선    10
Name: line, dtype: int64

# 지하철 역 주변 시설 정보 추가
- 위에서 만든 subway_information_total 데이터 셋에 다음과 같이 5개의 변수를 새롭게 추가해주겠다.
    1. 지하철 역 주변 중학교 개수
    2. 지하철 역 주변 고등학교 개수
    3. 지하철 역 주변 대학교 개수
    4. 지하철 역 주변 종합병원 개수
    5. 지하철 역 주변 영화관 개수
- 각 변수들을 추가하기 위해 사용한 데이터 셋은 다음과 같이 총 5개이다.
    1. 서울특별시 중학교 현황 데이터 셋
    2. 서울특별시 고등학교 현황 데이터 셋
    3. 서울특별시 대학교 현황 데이터 셋
    4. 서울특별시 종합병원 현황 데이터 셋
    5. 서울특별시 영화관 현황 데이터 셋

In [33]:
df_msi = pd.read_csv('{}/서울특별시_중학교_현황.csv'.format(filepath))
df_hsi = pd.read_csv('{}/서울특별시_고등학교_현황.csv'.format(filepath))
df_uvi = pd.read_csv('{}/서울특별시_대학교_현황.csv'.format(filepath))
df_hpi = pd.read_csv('{}/서울특별시_종합병원_현황.csv'.format(filepath))
df_mti = pd.read_csv('{}/서울특별시_영화관_현황.csv'.format(filepath))

## 지하철 역별 주변 중학교 개수

In [34]:
df_msi

호선      지하철명  환승역   교육청   소재구     학교명  설립  \
0     2        문래  NaN    남부  영등포구   양화중학교  공립   
1     2     영등포구청  2,5    남부  영등포구   양화중학교  공립   
2     5     영등포구청  2,5    남부  영등포구   양화중학교  공립   
3     5      여의나루  NaN    남부  영등포구  여의도중학교  공립   
4     2  대림(구로구청)  2,7    남부  영등포구   영남중학교  공립   
..   ..       ...  ...   ...   ...     ...  ..   
329   8        문정  NaN  강동송파   송파구   가원중학교  공립   
330   5        명일  NaN  강동송파   강동구   강일중학교  공립   
331   5        거여  NaN  강동송파   송파구   거원중학교  공립   
332   5       상일동  NaN  강동송파   강동구   고덕중학교  공립   
333   5        고덕  NaN  강동송파   강동구   명일중학교  공립   

                                     주소  1학년  2학년  3학년  총학생수  정규 교원수 계  \
0        서울특별시 영등포구 당산로 56 (문래동3가양화중학교)  231  287  246   764        45   
1        서울특별시 영등포구 당산로 56 (문래동3가양화중학교)  231  287  246   764        45   
2        서울특별시 영등포구 당산로 56 (문래동3가양화중학교)  231  287  246   764        45   
3    서울특별시 영등포구 국제금융로7길 46 (여의도동여의도중학교)  328  334  349  1011        57   
4      서울특별시 영등포구 대림로31나길 17 (대림동영남중학교)  257  300  296   853        50   
..                                  ...  ...  ...  ...   ...       ...   
329   서울특별시 송파구 중대로10길 40-18 (가락동가원중학교)  230  268  264   762        46   
330      서울특별시 강동구 고덕로27길 12 (암사동강일중학교)  211  270  265   746        42   
331       서울특별시 송파구 양산로2길 30 (거여동거원중학교)  232  246  244   722        40   
332         서울 강동구 동남로82길 94 (고덕동고덕중학교)  191  258  249   698        42   
333    서울특별시 강동구 양재대로156길 57 (고덕동명일중학교)  265  317  314   896        52   

     기간제 교원수 계  직원수  총유동인구수   거리  
0            2    4     815  210  
1            2    4     815  500  
2            2    4     815  500  
3            9    4    1081  420  
4            8    4     915  480  
..         ...  ...     ...  ...  
329          4    4     816  577  
330          3    4     795  667  
331          5    4     771  695  
332          8    4     752  508  
333          9    4     961  583  

[334 rows x 17 columns]

- 변수명을 다음과 같이 변경해주겠다.
    - 호선 = line
    - 지하철명 = station_name
    - 학교명 = msi_cnt

In [35]:
df_msi = df_msi[['호선', '지하철명', '학교명']]
df_msi.rename(columns = {'호선': 'line', '지하철명': 'station_name', '학교명': 'msi_cnt'}, inplace = True)
df_msi.head()

C:\Users\Brian\anaconda3\lib\site-packages\pandas\core\frame.py:4125: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().rename(


line station_name msi_cnt
0     2           문래   양화중학교
1     2        영등포구청   양화중학교
2     5        영등포구청   양화중학교
3     5         여의나루  여의도중학교
4     2     대림(구로구청)   영남중학교

- 다음으로 숫자로 되어 있는 line 변수 뒤에 '호선'이라는 문자를 붙여주겠다.
    - ex) "1" >> "1호선"

In [36]:
for i in range(len(df_msi)):
    df_msi['line'][i] = str(df_msi['line'][i]) + '호선'

df_msi.line.value_counts()

<ipython-input-36-45e36c1fbd44>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_msi['line'][i] = str(df_msi['line'][i]) + '호선'
C:\Users\Brian\anaconda3\lib\site-packages\pandas\core\indexing.py:671: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_with_indexer(indexer, value)
C:\Users\Brian\anaconda3\lib\site-packages\pandas\core\series.py:1024: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self.loc[key] = value
C:\Users

5호선    70
2호선    67
7호선    60
6호선    46
3호선    38
4호선    35
8호선    10
1호선     8
Name: line, dtype: int64

- 중학교 이름에 상관없이 그냥 '중학교'의 개수가 중요하므로, 중학교 이름을 모두 '중학교'로 변경해주겠다.

In [37]:
df_msi['msi_cnt'] = '중학교'
df_msi.msi_cnt.value_counts()

<ipython-input-37-3754dfd1b5f5>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_msi['msi_cnt'] = '중학교'


중학교    334
Name: msi_cnt, dtype: int64

- 그리고 'line'과 'station_name' 변수를 기준으로 그룹화해서, 각 호선/역명별로 주변에 존재하는 중학교 개수를 세주겠다.

In [38]:
df_msi2 = df_msi.groupby(['line', 'station_name']).count().reset_index()
df_msi2

line station_name  msi_cnt
0    1호선          동대문        1
1    1호선          서울역        2
2    1호선           시청        1
3    1호선          신설동        1
4    1호선          제기동        1
..   ...          ...      ...
185  8호선           석촌        1
186  8호선           송파        1
187  8호선           암사        1
188  8호선     잠실(송파구청)        1
189  8호선           장지        1

[190 rows x 3 columns]

- 이제 subway_information_total 데이터 셋에 df_msi2 데이터 셋을 'station_name'과 'line'을 조인키로 left join 하겠다.
    - 이렇게 하면 역 주변에 중학교가 존재하는 지하철 역들은 msi_cnt 값이 존재할 것이고, 그렇지 않다면 NaN 값이 들어갈 것이다.
    - 다음으로 msi_cnt에 존재하는 NaN 값들을 모두 '0'으로 변경한 뒤, msi_cnt 변수의 타입을 int형으로 변경해주겠다.

In [39]:
subway_information_total2 = pd.merge(subway_information_total, df_msi2, how = 'left', on = ['station_name', 'line'])
subway_information_total2.msi_cnt.fillna(0, inplace = True)
subway_information_total2['msi_cnt'] = subway_information_total2['msi_cnt'].astype('int')
subway_information_total2

line station_name  waiting_area  platform_area    gu dong  \
0    3호선         가락시장      12557.98        3508.00   송파구  가락동   
1    8호선         가락시장       5110.63        2617.50   송파구  가락동   
2    7호선      가산디지털단지      10116.83        2434.70   금천구  가산동   
3    2호선           강남       4018.00        2374.00   강남구  역삼동   
4    7호선         강남구청       7987.99        2371.32   강남구  삼성동   
..   ...          ...           ...            ...   ...  ...   
272  3호선           홍제       7220.00        1876.00  서대문구  홍제동   
273  5호선           화곡       7144.92        1982.74   강서구  화곡동   
274  6호선  화랑대(서울여대입구)       7497.20        2610.02   노원구  공릉동   
275  4호선    회현(남대문시장)       5253.00        2702.00    중구  남창동   
276  6호선        효창공원앞       4771.62        2022.17   용산구  효창동   

                                address type  length floor transfer  \
0            서울특별시 송파구 가락동 184-23 가락시장역  상대식     205    B4      3,8   
1    서울특별시 송파구 가락동 184-23 가락시장역(3, 8호선)  상대식     125    B2      3,8   
2           서울특별시 금천구 가산동 54-3 가산디지털단지역  상대식     165    B4      1,7   
3                 서울특별시 강남구 역삼동 858 강남역  상대식     205    B2    2,신분당   
4            서울특별시 강남구 삼성동 111-44 강남구청역  상대식     165    B3     7,분당   
..                                  ...  ...     ...   ...      ...   
272            서울특별시 서대문구 홍제동 161-1 홍제역   섬식     205    B3      NaN   
273           서울특별시 강서구 화곡동 1089-54 화곡역   섬식     165    B2      NaN   
274            서울특별시 노원구 공릉동 285-2 화랑대역  상대식     165    B2      NaN   
275           서울특별시 중구 남창동 64-1 회현역 4호선   섬식     205    B4      NaN   
276             서울특별시 용산구 효창동 80 효창공원앞역   섬식     165    B3   6,경의중앙   

     transfer_num  entrance                        gate  msi_cnt  
0               2         2                       7, 8,        1  
1               2         7             1,2,2-1,3,4,5,6        1  
2               2         5                   2,3,4,5,6        0  
3               2         8  1, 2, 7, 8, 9, 10, 11, 12,        2  
4               2         5                 1,2,3,3-1,4        1  
..            ...       ...                         ...      ...  
272             0         4                1, 2, 3, 4,         3  
273             0         8             1,2,3,4,5,6,7,8        1  
274             0         7               1,2,3,4,5,6,7        2  
275             0         7               1,2,3,4,5,6,7        0  
276             2         4                     1,2,3,6        1  

[277 rows x 15 columns]

## 지하철 역별 주변 고등학교 개수

In [40]:
df_hsi

호선         지하철명  환승역  교육청  소재구              학교명  설립  \
0     8           송파  NaN  서울시  송파구           가락고등학교  공립   
1     5           방화  NaN  서울시  강서구         강서공업고등학교  공립   
2     2     구의(광진구청)  NaN  서울시  광진구  건국대학교사범대학부속고등학교  사립   
3     7           청담  NaN  서울시  강남구           경기고등학교  공립   
4     7  공릉(서울과학기술대)  NaN  서울시  노원구       경기기계공업고등학교  공립   
..   ..          ...  ...  ...  ...              ...  ..   
247   5          애오개  NaN  서울시   중구           환일고등학교  사립   
248   2           아현  NaN  서울시   중구           환일고등학교  사립   
249   2   충정로(경기대입구)  2,5  서울시   중구           환일고등학교  사립   
250   5   충정로(경기대입구)  2,5  서울시   중구           환일고등학교  사립   
251   2     삼성(무역센터)  NaN  서울시  강남구           휘문고등학교  사립   

                                                주소  1학년  2학년  3학년  총학생수  정규교원  \
0                     서울특별시 송파구 송이로 42 (송파동가락고등학교)  351  388  455  1194    78   
1                서울특별시 강서구 방화대로47길 9 (방화동강서공업고등학교)  208  197  216   621    57   
2    서울특별시 광진구 능동로 120  건국대학교사범대학부속고등학교 (화양동건국대학교)  475  457  510  1442    73   
3                   서울특별시 강남구 영동대로 643 (삼성동경기고등학교)  608  560  569  1737    90   
4                서울특별시 노원구 공릉로 264 (하계동경기기계공업고등학교)  441  408  413  1262   139   
..                                             ...  ...  ...  ...   ...   ...   
247               서울특별시 중구 환일길 47  환일중고등학교 (만리동2가)  370  389  393  1152    71   
248               서울특별시 중구 환일길 47  환일중고등학교 (만리동2가)  370  389  393  1152    71   
249               서울특별시 중구 환일길 47  환일중고등학교 (만리동2가)  370  389  393  1152    71   
250               서울특별시 중구 환일길 47  환일중고등학교 (만리동2가)  370  389  393  1152    71   
251                            서울특별시 강남구 역삼로   541  428  446  457  1331    69   

     기간제교원  직원  총유동인구수   거리  
0        6   6    1284  466  
1        6  10     694  330  
2        9   5    1529  610  
3        5   9    1841  380  
4        8  16    1425  680  
..     ...  ..     ...  ...  
247      2   4    1229  450  
248      2   4    1229  630  
249      2   4    1229  690  
250      2   4    1229  690  
251     16   7    1423  450  

[252 rows x 17 columns]

- 변수명을 다음과 같이 변경해주겠다.
    - 호선 = line
    - 지하철명 = station_name
    - 학교명 = hsi_cnt

In [41]:
df_hsi = df_hsi[['호선', '지하철명', '학교명']]
df_hsi.rename(columns = {'호선': 'line', '지하철명': 'station_name', '학교명': 'hsi_cnt'}, inplace = True)
df_hsi.head()

C:\Users\Brian\anaconda3\lib\site-packages\pandas\core\frame.py:4125: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().rename(


line station_name          hsi_cnt
0     8           송파           가락고등학교
1     5           방화         강서공업고등학교
2     2     구의(광진구청)  건국대학교사범대학부속고등학교
3     7           청담           경기고등학교
4     7  공릉(서울과학기술대)       경기기계공업고등학교

- 다음으로 숫자로 되어 있는 line 변수 뒤에 '호선'이라는 문자를 붙여주겠다.
    - ex) "1" >> "1호선"

In [42]:
for i in range(len(df_hsi)):
    df_hsi['line'][i] = str(df_hsi['line'][i]) + '호선'

df_hsi.line.value_counts()

<ipython-input-42-91ea6930db25>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_hsi['line'][i] = str(df_hsi['line'][i]) + '호선'
C:\Users\Brian\anaconda3\lib\site-packages\pandas\core\indexing.py:671: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_with_indexer(indexer, value)
C:\Users\Brian\anaconda3\lib\site-packages\pandas\core\series.py:1024: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self.loc[key] = value
C:\Users

5호선    58
2호선    44
6호선    39
7호선    37
4호선    32
3호선    29
8호선     7
1호선     6
Name: line, dtype: int64

- 고등학교 이름에 상관없이 그냥 '고등학교'의 개수가 중요하므로, 고등학교 이름을 모두 '고등학교'로 변경해주겠다.

In [43]:
df_hsi['hsi_cnt'] = '고등학교'
df_hsi.hsi_cnt.value_counts()

<ipython-input-43-36d1e7041cf4>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_hsi['hsi_cnt'] = '고등학교'


고등학교    252
Name: hsi_cnt, dtype: int64

- 그리고 'line'과 'station_name' 변수를 기준으로 그룹화해서, 각 호선/역명별로 주변에 존재하는 고등학교 개수를 세주겠다.

In [44]:
df_hsi2 = df_hsi.groupby(['line', 'station_name']).count().reset_index()
df_hsi2

line station_name  hsi_cnt
0    1호선          동대문        1
1    1호선          동묘앞        1
2    1호선          서울역        1
3    1호선           시청        2
4    1호선          신설동        1
..   ...          ...      ...
141  8호선         강동구청        1
142  8호선           문정        1
143  8호선           석촌        1
144  8호선           송파        3
145  8호선           장지        1

[146 rows x 3 columns]

- 이제 subway_information_total2 데이터 셋에 df_hsi2 데이터 셋을 'station_name'과 'line'을 조인키로 left join 하겠다.
    - 이렇게 하면 역 주변에 고등학교가 존재하는 지하철 역들은 hsi_cnt 값이 존재할 것이고, 그렇지 않다면 NaN 값이 들어갈 것이다.
    - 다음으로 hsi_cnt에 존재하는 NaN 값들을 모두 '0'으로 변경한 뒤, hsi_cnt 변수의 타입을 int형으로 변경해주겠다.

In [45]:
subway_information_total3 = pd.merge(subway_information_total2, df_hsi2, how = 'left', on = ['station_name', 'line'])
subway_information_total3.hsi_cnt.fillna(0, inplace = True)
subway_information_total3['hsi_cnt'] = subway_information_total3['hsi_cnt'].astype('int')
subway_information_total3

line station_name  waiting_area  platform_area    gu dong  \
0    3호선         가락시장      12557.98        3508.00   송파구  가락동   
1    8호선         가락시장       5110.63        2617.50   송파구  가락동   
2    7호선      가산디지털단지      10116.83        2434.70   금천구  가산동   
3    2호선           강남       4018.00        2374.00   강남구  역삼동   
4    7호선         강남구청       7987.99        2371.32   강남구  삼성동   
..   ...          ...           ...            ...   ...  ...   
272  3호선           홍제       7220.00        1876.00  서대문구  홍제동   
273  5호선           화곡       7144.92        1982.74   강서구  화곡동   
274  6호선  화랑대(서울여대입구)       7497.20        2610.02   노원구  공릉동   
275  4호선    회현(남대문시장)       5253.00        2702.00    중구  남창동   
276  6호선        효창공원앞       4771.62        2022.17   용산구  효창동   

                                address type  length floor transfer  \
0            서울특별시 송파구 가락동 184-23 가락시장역  상대식     205    B4      3,8   
1    서울특별시 송파구 가락동 184-23 가락시장역(3, 8호선)  상대식     125    B2      3,8   
2           서울특별시 금천구 가산동 54-3 가산디지털단지역  상대식     165    B4      1,7   
3                 서울특별시 강남구 역삼동 858 강남역  상대식     205    B2    2,신분당   
4            서울특별시 강남구 삼성동 111-44 강남구청역  상대식     165    B3     7,분당   
..                                  ...  ...     ...   ...      ...   
272            서울특별시 서대문구 홍제동 161-1 홍제역   섬식     205    B3      NaN   
273           서울특별시 강서구 화곡동 1089-54 화곡역   섬식     165    B2      NaN   
274            서울특별시 노원구 공릉동 285-2 화랑대역  상대식     165    B2      NaN   
275           서울특별시 중구 남창동 64-1 회현역 4호선   섬식     205    B4      NaN   
276             서울특별시 용산구 효창동 80 효창공원앞역   섬식     165    B3   6,경의중앙   

     transfer_num  entrance                        gate  msi_cnt  hsi_cnt  
0               2         2                       7, 8,        1        0  
1               2         7             1,2,2-1,3,4,5,6        1        0  
2               2         5                   2,3,4,5,6        0        0  
3               2         8  1, 2, 7, 8, 9, 10, 11, 12,        2        0  
4               2         5                 1,2,3,3-1,4        1        1  
..            ...       ...                         ...      ...      ...  
272             0         4                1, 2, 3, 4,         3        0  
273             0         8             1,2,3,4,5,6,7,8        1        0  
274             0         7               1,2,3,4,5,6,7        2        2  
275             0         7               1,2,3,4,5,6,7        0        0  
276             2         4                     1,2,3,6        1        1  

[277 rows x 16 columns]

## 지하철 역별 주변 대학교 개수

In [46]:
df_uvi

호선        지하철명          환승역           이름   캠퍼스  \
0     2          서초          NaN       가톨릭대학교  성의교정   
1     3       고속터미널        3,7,9       가톨릭대학교  성의교정   
2     7       고속터미널        3,7,9       가톨릭대학교  성의교정   
3     4          혜화          NaN       가톨릭대학교  성신교정   
4     4  한성대입구(삼선교)          NaN       가톨릭대학교  성신교정   
..   ..         ...          ...          ...   ...   
147   2          선릉         2,분당   한림국제대학원대학교   NaN   
148   4    숙대입구(갈월)          NaN  한반도국제대학원대학교   NaN   
149   5          공덕  5,6,경의중앙,공항  한반도국제대학원대학교   NaN   
150   6          공덕  5,6,경의중앙,공항  한반도국제대학원대학교   NaN   
151   6       효창공원앞       6,경의중앙  한반도국제대학원대학교   NaN   

                                                  홈페이지 설립유형   위치  \
0        https://www.catholic.ac.kr/www/about51_3.html   사립  서초구   
1        https://www.catholic.ac.kr/www/about51_3.html   사립  서초구   
2        https://www.catholic.ac.kr/www/about51_3.html   사립  서초구   
3        https://www.catholic.ac.kr/www/about51_4.html   사립  종로구   
4        https://www.catholic.ac.kr/www/about51_4.html   사립  종로구   
..                                                 ...  ...  ...   
147  http://www.hugs.ac.kr:18080/site/graduate/dire...   사립  강남구   
148  http://www.kuis.ac.kr/bbs/board.php?bo_table=h...   사립  용산구   
149  http://www.kuis.ac.kr/bbs/board.php?bo_table=h...   사립  용산구   
150  http://www.kuis.ac.kr/bbs/board.php?bo_table=h...   사립  용산구   
151  http://www.kuis.ac.kr/bbs/board.php?bo_table=h...   사립  용산구   

                   규모   대학유형  
0    3캠퍼스, 9대학, 14대학원     대학  
1    3캠퍼스, 9대학, 14대학원     대학  
2    3캠퍼스, 9대학, 14대학원     대학  
3    3캠퍼스, 9대학, 14대학원     대학  
4    3캠퍼스, 9대학, 14대학원     대학  
..                ...    ...  
147               NaN  특수대학원  
148               NaN  특수대학원  
149               NaN  특수대학원  
150               NaN  특수대학원  
151               NaN  특수대학원  

[152 rows x 10 columns]

- 변수명을 다음과 같이 변경해주겠다.
    - 호선 = line
    - 지하철명 = station_name
    - 이름 = uvi_cnt

In [47]:
df_uvi = df_uvi[['호선', '지하철명', '이름']]
df_uvi.rename(columns = {'호선': 'line', '지하철명': 'station_name', '이름': 'uvi_cnt'}, inplace = True)
df_uvi.head()

C:\Users\Brian\anaconda3\lib\site-packages\pandas\core\frame.py:4125: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().rename(


line station_name uvi_cnt
0     2           서초  가톨릭대학교
1     3        고속터미널  가톨릭대학교
2     7        고속터미널  가톨릭대학교
3     4           혜화  가톨릭대학교
4     4   한성대입구(삼선교)  가톨릭대학교

- 다음으로 숫자로 되어 있는 line 변수 뒤에 '호선'이라는 문자를 붙여주겠다.
    - ex) "1" >> "1호선"

In [48]:
for i in range(len(df_uvi)):
    df_uvi['line'][i] = str(df_uvi['line'][i]) + '호선'

df_uvi.line.value_counts()

<ipython-input-48-f9f3ed79b1e6>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_uvi['line'][i] = str(df_uvi['line'][i]) + '호선'
C:\Users\Brian\anaconda3\lib\site-packages\pandas\core\indexing.py:671: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_with_indexer(indexer, value)
C:\Users\Brian\anaconda3\lib\site-packages\pandas\core\series.py:1024: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self.loc[key] = value
C:\Users

2호선    30
6호선    28
4호선    24
3호선    23
7호선    21
5호선    18
1호선     7
8호선     1
Name: line, dtype: int64

- 대학교 이름에 상관없이 그냥 '대학교'의 개수가 중요하므로, 대학교 이름을 모두 '대학교'로 변경해주겠다.

In [49]:
df_uvi['uvi_cnt'] = '대학교'
df_uvi.uvi_cnt.value_counts()

<ipython-input-49-2727dae3a378>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_uvi['uvi_cnt'] = '대학교'


대학교    152
Name: uvi_cnt, dtype: int64

- 그리고 'line'과 'station_name' 변수를 기준으로 그룹화해서, 각 호선/역명별로 주변에 존재하는 대학교 개수를 세주겠다.

In [50]:
df_uvi2 = df_uvi.groupby(['line', 'station_name']).count().reset_index()
df_uvi2

line  station_name  uvi_cnt
0   1호선            종각        2
1   1호선          종로3가        3
2   1호선  청량리(서울시립대입구)        2
3   2호선          건대입구        1
4   2호선    교대(법원.검찰청)        1
..  ...           ...      ...
94  7호선            중계        1
95  7호선          태릉입구        3
96  7호선            하계        1
97  7호선            학동        1
98  8호선          강동구청        1

[99 rows x 3 columns]

- 이제 subway_information_total3 데이터 셋에 df_uvi2 데이터 셋을 'station_name'과 'line'을 조인키로 left join 하겠다.
    - 이렇게 하면 역 주변에 대학교가 존재하는 지하철 역들은 uvi_cnt 값이 존재할 것이고, 그렇지 않다면 NaN 값이 들어갈 것이다.
    - 다음으로 uvi_cnt에 존재하는 NaN 값들을 모두 '0'으로 변경한 뒤, uvi_cnt 변수의 타입을 int형으로 변경해주겠다.

In [51]:
subway_information_total4 = pd.merge(subway_information_total3, df_uvi2, how = 'left', on = ['station_name', 'line'])
subway_information_total4.uvi_cnt.fillna(0, inplace = True)
subway_information_total4['uvi_cnt'] = subway_information_total4['uvi_cnt'].astype('int')
subway_information_total4

line station_name  waiting_area  platform_area    gu dong  \
0    3호선         가락시장      12557.98        3508.00   송파구  가락동   
1    8호선         가락시장       5110.63        2617.50   송파구  가락동   
2    7호선      가산디지털단지      10116.83        2434.70   금천구  가산동   
3    2호선           강남       4018.00        2374.00   강남구  역삼동   
4    7호선         강남구청       7987.99        2371.32   강남구  삼성동   
..   ...          ...           ...            ...   ...  ...   
272  3호선           홍제       7220.00        1876.00  서대문구  홍제동   
273  5호선           화곡       7144.92        1982.74   강서구  화곡동   
274  6호선  화랑대(서울여대입구)       7497.20        2610.02   노원구  공릉동   
275  4호선    회현(남대문시장)       5253.00        2702.00    중구  남창동   
276  6호선        효창공원앞       4771.62        2022.17   용산구  효창동   

                                address type  length floor transfer  \
0            서울특별시 송파구 가락동 184-23 가락시장역  상대식     205    B4      3,8   
1    서울특별시 송파구 가락동 184-23 가락시장역(3, 8호선)  상대식     125    B2      3,8   
2           서울특별시 금천구 가산동 54-3 가산디지털단지역  상대식     165    B4      1,7   
3                 서울특별시 강남구 역삼동 858 강남역  상대식     205    B2    2,신분당   
4            서울특별시 강남구 삼성동 111-44 강남구청역  상대식     165    B3     7,분당   
..                                  ...  ...     ...   ...      ...   
272            서울특별시 서대문구 홍제동 161-1 홍제역   섬식     205    B3      NaN   
273           서울특별시 강서구 화곡동 1089-54 화곡역   섬식     165    B2      NaN   
274            서울특별시 노원구 공릉동 285-2 화랑대역  상대식     165    B2      NaN   
275           서울특별시 중구 남창동 64-1 회현역 4호선   섬식     205    B4      NaN   
276             서울특별시 용산구 효창동 80 효창공원앞역   섬식     165    B3   6,경의중앙   

     transfer_num  entrance                        gate  msi_cnt  hsi_cnt  \
0               2         2                       7, 8,        1        0   
1               2         7             1,2,2-1,3,4,5,6        1        0   
2               2         5                   2,3,4,5,6        0        0   
3               2         8  1, 2, 7, 8, 9, 10, 11, 12,        2        0   
4               2         5                 1,2,3,3-1,4        1        1   
..            ...       ...                         ...      ...      ...   
272             0         4                1, 2, 3, 4,         3        0   
273             0         8             1,2,3,4,5,6,7,8        1        0   
274             0         7               1,2,3,4,5,6,7        2        2   
275             0         7               1,2,3,4,5,6,7        0        0   
276             2         4                     1,2,3,6        1        1   

     uvi_cnt  
0          0  
1          0  
2          0  
3          0  
4          0  
..       ...  
272        1  
273        1  
274        3  
275        1  
276        2  

[277 rows x 17 columns]

## 지하철 역별 주변 종합병원 개수

In [52]:
df_hpi

호선         지하철명  환승역   소재구        병원명                       주소   거리
0    2  서울대입구(관악구청)  NaN   관악구     강남고려병원    서울 관악구 관악로 242 강남고려병원  580
1    7           신풍  NaN  영등포구     강남성심병원     서울 영등포구 신길로 1 강남성심병원  700
2    3           매봉  NaN   강남구   강남세브란스병원  서울 강남구 언주로 211 강남세브란스병원  650
3    2           역삼  NaN   강남구      강남차병원       서울 강남구 논현로 566 차병원  680
4    5           고덕  NaN   강동구  강동경희대학교병원           서울 강동구 동남로 892  340
..  ..          ...  ...   ...        ...                      ...  ...
79   5           신길  1,5  영등포구     한강성심병원         서울 영등포구 버드나루로 55  700
80   2          한양대  NaN   성동구    한양대학교병원        서울 성동구 왕십리로 222-1  330
81   4           쌍문  NaN   도봉구       한전병원     서울 도봉구 우이천로 308 한일병원  580
82   2     구의(광진구청)  NaN   광진구       혜민병원       서울 광진구 자양로 85 혜민병원  310
83   5           목동  NaN   양천구       홍익병원    서울 양천구 목동로 225 홍익병원본관  290

[84 rows x 7 columns]

- 변수명을 다음과 같이 변경해주겠다.
    - 호선 = line
    - 지하철명 = station_name
    - 병원명 = hpi_cnt

In [53]:
df_hpi = df_hpi[['호선', '지하철명', '병원명']]
df_hpi.rename(columns = {'호선': 'line', '지하철명': 'station_name', '병원명': 'hpi_cnt'}, inplace = True)
df_hpi.head()

C:\Users\Brian\anaconda3\lib\site-packages\pandas\core\frame.py:4125: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().rename(


line station_name    hpi_cnt
0     2  서울대입구(관악구청)     강남고려병원
1     7           신풍     강남성심병원
2     3           매봉   강남세브란스병원
3     2           역삼      강남차병원
4     5           고덕  강동경희대학교병원

- 다음으로 숫자로 되어 있는 line 변수 뒤에 '호선'이라는 문자를 붙여주겠다.
    - ex) "1" >> "1호선"

In [54]:
for i in range(len(df_hpi)):
    df_hpi['line'][i] = str(df_hpi['line'][i]) + '호선'

df_hpi.line.value_counts()

<ipython-input-54-6bc40f99fc77>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_hpi['line'][i] = str(df_hpi['line'][i]) + '호선'
C:\Users\Brian\anaconda3\lib\site-packages\pandas\core\indexing.py:671: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_with_indexer(indexer, value)
C:\Users\Brian\anaconda3\lib\site-packages\pandas\core\series.py:1024: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self.loc[key] = value
C:\Users

2호선    21
5호선    18
7호선    13
3호선    12
4호선    10
1호선     6
6호선     4
Name: line, dtype: int64

- 종합병원 이름에 상관없이 그냥 '종합병원'의 개수가 중요하므로, 종합병원 이름을 모두 '종합병원'으로 변경해주겠다.

In [55]:
df_hpi['hpi_cnt'] = '종합병원'
df_hpi.hpi_cnt.value_counts()

<ipython-input-55-d15a95b50ab4>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_hpi['hpi_cnt'] = '종합병원'


종합병원    84
Name: hpi_cnt, dtype: int64

- 그리고 'line'과 'station_name' 변수를 기준으로 그룹화해서, 각 호선/역명별로 주변에 존재하는 종합병원 개수를 세주겠다.

In [56]:
df_hpi2 = df_hpi.groupby(['line', 'station_name']).count().reset_index()
df_hpi2

line  station_name  hpi_cnt
0   1호선           동대문        1
1   1호선           신설동        1
2   1호선           제기동        1
3   1호선          종로3가        1
4   1호선  청량리(서울시립대입구)        2
..  ...           ...      ...
68  7호선        신대방삼거리        1
69  7호선            신풍        1
70  7호선            중계        1
71  7호선            청담        1
72  7호선            학동        1

[73 rows x 3 columns]

- 이제 subway_information_total4 데이터 셋에 df_hpi2 데이터 셋을 'station_name'과 'line'을 조인키로 left join 하겠다.
    - 이렇게 하면 역 주변에 종합병원이 존재하는 지하철 역들은 hpi_cnt 값이 존재할 것이고, 그렇지 않다면 NaN 값이 들어갈 것이다.
    - 다음으로 hpi_cnt에 존재하는 NaN 값들을 모두 '0'으로 변경한 뒤, hpi_cnt 변수의 타입을 int형으로 변경해주겠다.

In [57]:
subway_information_total5 = pd.merge(subway_information_total4, df_hpi2, how = 'left', on = ['station_name', 'line'])
subway_information_total5.hpi_cnt.fillna(0, inplace = True)
subway_information_total5['hpi_cnt'] = subway_information_total5['hpi_cnt'].astype('int')
subway_information_total5

line station_name  waiting_area  platform_area    gu dong  \
0    3호선         가락시장      12557.98        3508.00   송파구  가락동   
1    8호선         가락시장       5110.63        2617.50   송파구  가락동   
2    7호선      가산디지털단지      10116.83        2434.70   금천구  가산동   
3    2호선           강남       4018.00        2374.00   강남구  역삼동   
4    7호선         강남구청       7987.99        2371.32   강남구  삼성동   
..   ...          ...           ...            ...   ...  ...   
272  3호선           홍제       7220.00        1876.00  서대문구  홍제동   
273  5호선           화곡       7144.92        1982.74   강서구  화곡동   
274  6호선  화랑대(서울여대입구)       7497.20        2610.02   노원구  공릉동   
275  4호선    회현(남대문시장)       5253.00        2702.00    중구  남창동   
276  6호선        효창공원앞       4771.62        2022.17   용산구  효창동   

                                address type  length floor transfer  \
0            서울특별시 송파구 가락동 184-23 가락시장역  상대식     205    B4      3,8   
1    서울특별시 송파구 가락동 184-23 가락시장역(3, 8호선)  상대식     125    B2      3,8   
2           서울특별시 금천구 가산동 54-3 가산디지털단지역  상대식     165    B4      1,7   
3                 서울특별시 강남구 역삼동 858 강남역  상대식     205    B2    2,신분당   
4            서울특별시 강남구 삼성동 111-44 강남구청역  상대식     165    B3     7,분당   
..                                  ...  ...     ...   ...      ...   
272            서울특별시 서대문구 홍제동 161-1 홍제역   섬식     205    B3      NaN   
273           서울특별시 강서구 화곡동 1089-54 화곡역   섬식     165    B2      NaN   
274            서울특별시 노원구 공릉동 285-2 화랑대역  상대식     165    B2      NaN   
275           서울특별시 중구 남창동 64-1 회현역 4호선   섬식     205    B4      NaN   
276             서울특별시 용산구 효창동 80 효창공원앞역   섬식     165    B3   6,경의중앙   

     transfer_num  entrance                        gate  msi_cnt  hsi_cnt  \
0               2         2                       7, 8,        1        0   
1               2         7             1,2,2-1,3,4,5,6        1        0   
2               2         5                   2,3,4,5,6        0        0   
3               2         8  1, 2, 7, 8, 9, 10, 11, 12,        2        0   
4               2         5                 1,2,3,3-1,4        1        1   
..            ...       ...                         ...      ...      ...   
272             0         4                1, 2, 3, 4,         3        0   
273             0         8             1,2,3,4,5,6,7,8        1        0   
274             0         7               1,2,3,4,5,6,7        2        2   
275             0         7               1,2,3,4,5,6,7        0        0   
276             2         4                     1,2,3,6        1        1   

     uvi_cnt  hpi_cnt  
0          0        0  
1          0        0  
2          0        0  
3          0        0  
4          0        0  
..       ...      ...  
272        1        0  
273        1        0  
274        3        0  
275        1        0  
276        2        0  

[277 rows x 18 columns]

## 지하철 역별 주변 영화관 개수
- CGV, 롯데시네마, 메가박스만 포함하겠다.

In [58]:
df_mti

호선     전철역명    환승역 광역단체  기초단체         영화상영관명  총스크린수  \
0     2      신도림    1,2  서울시   구로구        씨네큐 신도림     10   
1     3       매봉    NaN  서울시   강남구       롯데시네마 도곡      7   
2     4       혜화    NaN  서울시   종로구        CGV 대학로      8   
3     2  구로디지털단지    NaN  서울시   구로구         CGV 구로     10   
4     2       신촌    NaN  서울시  서대문구     CGV 신촌아트레온     10   
..   ..      ...    ...  ...   ...            ...    ...   
99    1     종로3가  1,3,5  서울시   종로구  CGV 피카디리 1958      8   
100   3     종로3가  1,3,5  서울시   종로구  CGV 피카디리 1958      8   
101   5     종로3가  1,3,5  서울시   종로구  CGV 피카디리 1958      8   
102   3      연신내    3,6  서울시   은평구        메가박스 은평      6   
103   6      연신내    3,6  서울시   은평구        메가박스 은평      6   

                                         주소  
0    서울특별시 구로구 구로동 3-25 번지 신도림테크노마트 11층~14층  
1           서울특별시 강남구 도곡동 174-3 번지 롯데시네마 도곡  
2          서울특별시 종로구 명륜2가 41-9 번지 판타지움빌딩 1층  
3                      서울특별시 구로구 구로동 573 번지  
4                   서울특별시 서대문구 창천동 20-25 번지  
..                                      ...  
99                     서울특별시 종로구 돈의동 137 번지  
100                    서울특별시 종로구 돈의동 137 번지  
101                    서울특별시 종로구 돈의동 137 번지  
102   서울특별시 은평구 불광2동 304-2 번지 파레제페 빌딩 5층-8층  
103   서울특별시 은평구 불광2동 304-2 번지 파레제페 빌딩 5층-8층  

[104 rows x 8 columns]

- 변수명을 다음과 같이 변경해주겠다.
    - 호선 = line
    - 전철역명 = station_name
    - 영화상영관명 = mti_cnt

In [59]:
df_mti = df_mti[['호선', '전철역명', '영화상영관명']]
df_mti.rename(columns = {'호선': 'line', '전철역명': 'station_name', '영화상영관명': 'mti_cnt'}, inplace = True)
df_mti.head()

C:\Users\Brian\anaconda3\lib\site-packages\pandas\core\frame.py:4125: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().rename(


line station_name     mti_cnt
0     2          신도림     씨네큐 신도림
1     3           매봉    롯데시네마 도곡
2     4           혜화     CGV 대학로
3     2      구로디지털단지      CGV 구로
4     2           신촌  CGV 신촌아트레온

- 다음으로 숫자로 되어 있는 line 변수 뒤에 '호선'이라는 문자를 붙여주겠다.
    - ex) "1" >> "1호선"

In [60]:
for i in range(len(df_mti)):
    df_mti['line'][i] = str(df_mti['line'][i]) + '호선'

df_mti.line.value_counts()

<ipython-input-60-07bc74adfb8f>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_mti['line'][i] = str(df_mti['line'][i]) + '호선'
C:\Users\Brian\anaconda3\lib\site-packages\pandas\core\indexing.py:671: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_with_indexer(indexer, value)
C:\Users\Brian\anaconda3\lib\site-packages\pandas\core\series.py:1024: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self.loc[key] = value
C:\Users

2호선    30
4호선    18
5호선    16
7호선    12
3호선     9
6호선     7
8호선     6
1호선     6
Name: line, dtype: int64

- 영화관 이름이 "CGV, 롯데시네마, 메가박스"에 해당하는 자료만 사용하도록 하겠다.

In [61]:
df_CGV = df_mti[df_mti['mti_cnt'].str.contains('CGV')].reset_index(drop = True)
df_lotte = df_mti[df_mti['mti_cnt'].str.contains('롯데시네마')].reset_index(drop = True)
df_megabox = df_mti[df_mti['mti_cnt'].str.contains('메가박스')].reset_index(drop = True)

len(df_CGV), len(df_lotte), len(df_megabox)

(38, 40, 23)

In [62]:
df_mti2 = pd.concat([df_CGV, df_lotte, df_megabox], axis = 0)
df_mti2

line station_name        mti_cnt
0   4호선           혜화        CGV 대학로
1   2호선      구로디지털단지         CGV 구로
2   2호선           신촌     CGV 신촌아트레온
3   2호선           이대     CGV 신촌아트레온
4   3호선           불광         CGV 불광
..  ...          ...            ...
18  7호선       군자(능동)         메가박스군자
19  8호선         강동구청         메가박스강동
20  2호선           강남  메가박스 씨티(강남대로)
21  3호선          연신내        메가박스 은평
22  6호선          연신내        메가박스 은평

[101 rows x 3 columns]

- 영화관 이름에 상관없이 그냥 '영화관'의 개수가 중요하므로, 영화관 이름을 모두 '영화관'으로 변경해주겠다.

In [63]:
df_mti2['mti_cnt'] = '영화관'
df_mti2.mti_cnt.value_counts()

영화관    101
Name: mti_cnt, dtype: int64

- 그리고 'line'과 'station_name' 변수를 기준으로 그룹화해서, 각 호선/역명별로 주변에 존재하는 영화관 개수를 세주겠다.

In [64]:
df_mti3 = df_mti2.groupby(['line', 'station_name']).count().reset_index()
df_mti3

line station_name  mti_cnt
0   1호선          동대문        1
1   1호선          동묘앞        1
2   1호선           시청        1
3   1호선          신설동        1
4   1호선         종로3가        1
..  ...          ...      ...
74  7호선           하계        2
75  8호선         강동구청        1
76  8호선     잠실(송파구청)        2
77  8호선           장지        2
78  8호선     천호(풍납토성)        1

[79 rows x 3 columns]

- 이제 subway_information_total5 데이터 셋에 df_mti3 데이터 셋을 'station_name'과 'line'을 조인키로 left join 하겠다.
    - 이렇게 하면 역 주변에 영화관이 존재하는 지하철 역들은 mti_cnt 값이 존재할 것이고, 그렇지 않다면 NaN 값이 들어갈 것이다.
    - 다음으로 mti_cnt에 존재하는 NaN 값들을 모두 '0'으로 변경한 뒤, mti_cnt 변수의 타입을 int형으로 변경해주겠다.

In [65]:
subway_information_total6 = pd.merge(subway_information_total5, df_mti3, how = 'left', on = ['station_name', 'line'])
subway_information_total6.mti_cnt.fillna(0, inplace = True)
subway_information_total6['mti_cnt'] = subway_information_total6['mti_cnt'].astype('int')
subway_information_total6

line station_name  waiting_area  platform_area    gu dong  \
0    3호선         가락시장      12557.98        3508.00   송파구  가락동   
1    8호선         가락시장       5110.63        2617.50   송파구  가락동   
2    7호선      가산디지털단지      10116.83        2434.70   금천구  가산동   
3    2호선           강남       4018.00        2374.00   강남구  역삼동   
4    7호선         강남구청       7987.99        2371.32   강남구  삼성동   
..   ...          ...           ...            ...   ...  ...   
272  3호선           홍제       7220.00        1876.00  서대문구  홍제동   
273  5호선           화곡       7144.92        1982.74   강서구  화곡동   
274  6호선  화랑대(서울여대입구)       7497.20        2610.02   노원구  공릉동   
275  4호선    회현(남대문시장)       5253.00        2702.00    중구  남창동   
276  6호선        효창공원앞       4771.62        2022.17   용산구  효창동   

                                address type  length floor transfer  \
0            서울특별시 송파구 가락동 184-23 가락시장역  상대식     205    B4      3,8   
1    서울특별시 송파구 가락동 184-23 가락시장역(3, 8호선)  상대식     125    B2      3,8   
2           서울특별시 금천구 가산동 54-3 가산디지털단지역  상대식     165    B4      1,7   
3                 서울특별시 강남구 역삼동 858 강남역  상대식     205    B2    2,신분당   
4            서울특별시 강남구 삼성동 111-44 강남구청역  상대식     165    B3     7,분당   
..                                  ...  ...     ...   ...      ...   
272            서울특별시 서대문구 홍제동 161-1 홍제역   섬식     205    B3      NaN   
273           서울특별시 강서구 화곡동 1089-54 화곡역   섬식     165    B2      NaN   
274            서울특별시 노원구 공릉동 285-2 화랑대역  상대식     165    B2      NaN   
275           서울특별시 중구 남창동 64-1 회현역 4호선   섬식     205    B4      NaN   
276             서울특별시 용산구 효창동 80 효창공원앞역   섬식     165    B3   6,경의중앙   

     transfer_num  entrance                        gate  msi_cnt  hsi_cnt  \
0               2         2                       7, 8,        1        0   
1               2         7             1,2,2-1,3,4,5,6        1        0   
2               2         5                   2,3,4,5,6        0        0   
3               2         8  1, 2, 7, 8, 9, 10, 11, 12,        2        0   
4               2         5                 1,2,3,3-1,4        1        1   
..            ...       ...                         ...      ...      ...   
272             0         4                1, 2, 3, 4,         3        0   
273             0         8             1,2,3,4,5,6,7,8        1        0   
274             0         7               1,2,3,4,5,6,7        2        2   
275             0         7               1,2,3,4,5,6,7        0        0   
276             2         4                     1,2,3,6        1        1   

     uvi_cnt  hpi_cnt  mti_cnt  
0          0        0        0  
1          0        0        0  
2          0        0        1  
3          0        0        3  
4          0        0        0  
..       ...      ...      ...  
272        1        0        0  
273        1        0        1  
274        3        0        0  
275        1        0        0  
276        2        0        1  

[277 rows x 19 columns]

# subway_boarding_total2 데이터 셋과 subway_information_total6 데이터 셋 결합
- 마지막으로 "2018~2019년 지하철 승하차 인원 데이터 셋(subway_boarding_total2)"과 "지하철 역 및 주변 시설 정보 데이터 셋(subway_information_total6)"을 결합해주겠다.

In [66]:
subway_semi_final = pd.merge(subway_boarding_total2, subway_information_total6, how = 'inner', on = ['station_name', 'line'])
subway_semi_final

date line  station_num station_name boarding  hour  \
0       2018-01-01  1호선          150          서울역       승차     0   
1       2018-01-01  1호선          150          서울역       하차     0   
2       2018-01-01  1호선          150          서울역       승차     5   
3       2018-01-01  1호선          150          서울역       하차     5   
4       2018-01-01  1호선          150          서울역       승차     6   
...            ...  ...          ...          ...      ...   ...   
8031075 2019-12-28  6호선         2615          연신내       하차    21   
8031076 2019-12-28  6호선         2615          연신내       승차    22   
8031077 2019-12-28  6호선         2615          연신내       하차    22   
8031078 2019-12-28  6호선         2615          연신내       승차    23   
8031079 2019-12-28  6호선         2615          연신내       하차    23   

         boarding_num  waiting_area  platform_area   gu  ... floor   transfer  \
0                  96        8725.0        2080.00   중구  ...    B2  1,4,공항,경의   
1                 147        8725.0        2080.00   중구  ...    B2  1,4,공항,경의   
2                 373        8725.0        2080.00   중구  ...    B2  1,4,공항,경의   
3                 205        8725.0        2080.00   중구  ...    B2  1,4,공항,경의   
4                 318        8725.0        2080.00   중구  ...    B2  1,4,공항,경의   
...               ...           ...            ...  ...  ...   ...        ...   
8031075             0        3241.0        2419.54  은평구  ...    B5        3,6   
8031076             0        3241.0        2419.54  은평구  ...    B5        3,6   
8031077             0        3241.0        2419.54  은평구  ...    B5        3,6   
8031078             0        3241.0        2419.54  은평구  ...    B5        3,6   
8031079             0        3241.0        2419.54  은평구  ...    B5        3,6   

        transfer_num  entrance                        gate msi_cnt  hsi_cnt  \
0                  4         9  1, 2, 3, 4, 5, 6, 7, 8, 9,       2        1   
1                  4         9  1, 2, 3, 4, 5, 6, 7, 8, 9,       2        1   
2                  4         9  1, 2, 3, 4, 5, 6, 7, 8, 9,       2        1   
3                  4         9  1, 2, 3, 4, 5, 6, 7, 8, 9,       2        1   
4                  4         9  1, 2, 3, 4, 5, 6, 7, 8, 9,       2        1   
...              ...       ...                         ...     ...      ...   
8031075            2         0                         NaN       2        2   
8031076            2         0                         NaN       2        2   
8031077            2         0                         NaN       2        2   
8031078            2         0                         NaN       2        2   
8031079            2         0                         NaN       2        2   

         uvi_cnt hpi_cnt  mti_cnt  
0              0       0        0  
1              0       0        0  
2              0       0        0  
3              0       0        0  
4              0       0        0  
...          ...     ...      ...  
8031075        0       1        1  
8031076        0       1        1  
8031077        0       1        1  
8031078        0       1        1  
8031079        0       1        1  

[8031080 rows x 24 columns]

In [67]:
subway_semi_final.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 8031080 entries, 0 to 8031079
Data columns (total 24 columns):
 #   Column         Dtype         
---  ------         -----         
 0   date           datetime64[ns]
 1   line           object        
 2   station_num    int64         
 3   station_name   object        
 4   boarding       object        
 5   hour           int64         
 6   boarding_num   int64         
 7   waiting_area   float64       
 8   platform_area  float64       
 9   gu             object        
 10  dong           object        
 11  address        object        
 12  type           object        
 13  length         int64         
 14  floor          object        
 15  transfer       object        
 16  transfer_num   int64         
 17  entrance       int64         
 18  gate           object        
 19  msi_cnt        int32         
 20  hsi_cnt        int32         
 21  uvi_cnt        int32         
 22  hpi_cnt        int32         
 23  mti_cnt

# '법정동'을 '행정동'으로 변경
- 여기까지 해서 생성된 지하철 데이터 셋의 '동(dong)' 변수에는 **"법정동"**이 입력되어 있다.
- 그런데 추후 결합해주어야 할 인구 데이터 셋의 '동(dong)' 변수에는 **"행정동"**이 입력되어 있다.
    - 이 부분을 해결하기 위해 네이버 지도를 기반으로 "지하철 역 주소의 행정동"을 직접 입력한 데이터 셋을 만들어주었다.
        - 생성된 파일명은 "지하철 역 주소_행정동.csv"
- 따라서 subway_semi_final 데이터 셋에 "지하철 역 주소_행정동.csv" 파일을 결합해주겠다.
    - subway_semi_final 데이터 셋에 존재하는 기존의 dong 변수를 제거하고, 결합을 통해 새로운 dong 변수를 만들어주겠다.

In [68]:
df_dong = pd.read_csv('{}/지하철 역 주소_행정동.csv'.format(filepath))
df_dong

line      station_name         dong
0    1호선               서울역          회현동
1    1호선                시청          소공동
2    1호선                종각  종로1.2.3.4가동
3    1호선              종로3가  종로1.2.3.4가동
4    1호선              종로5가      종로5.6가동
..   ...               ...          ...
272  8호선  남한산성입구(성남법원.검찰청)         금광2동
273  8호선             단대오거리         신흥2동
274  8호선                신흥         신흥3동
275  8호선                수진         수진1동
276  8호선                모란         수진2동

[277 rows x 3 columns]

In [69]:
# 기존의 dong 변수 제거
subway_semi_final.drop(['dong'], axis = 1, inplace = True)
subway_semi_final.shape

(8031080, 23)

In [70]:
# 데이터 셋 결합을 통해 새로운 dong 변수 생성
subway_final = pd.merge(subway_semi_final, df_dong, how = 'inner', on = ['station_name', 'line'])
subway_final

date line  station_num station_name boarding  hour  \
0       2018-01-01  1호선          150          서울역       승차     0   
1       2018-01-01  1호선          150          서울역       하차     0   
2       2018-01-01  1호선          150          서울역       승차     5   
3       2018-01-01  1호선          150          서울역       하차     5   
4       2018-01-01  1호선          150          서울역       승차     6   
...            ...  ...          ...          ...      ...   ...   
8031075 2019-12-28  6호선         2615          연신내       하차    21   
8031076 2019-12-28  6호선         2615          연신내       승차    22   
8031077 2019-12-28  6호선         2615          연신내       하차    22   
8031078 2019-12-28  6호선         2615          연신내       승차    23   
8031079 2019-12-28  6호선         2615          연신내       하차    23   

         boarding_num  waiting_area  platform_area   gu  ...   transfer  \
0                  96        8725.0        2080.00   중구  ...  1,4,공항,경의   
1                 147        8725.0        2080.00   중구  ...  1,4,공항,경의   
2                 373        8725.0        2080.00   중구  ...  1,4,공항,경의   
3                 205        8725.0        2080.00   중구  ...  1,4,공항,경의   
4                 318        8725.0        2080.00   중구  ...  1,4,공항,경의   
...               ...           ...            ...  ...  ...        ...   
8031075             0        3241.0        2419.54  은평구  ...        3,6   
8031076             0        3241.0        2419.54  은평구  ...        3,6   
8031077             0        3241.0        2419.54  은평구  ...        3,6   
8031078             0        3241.0        2419.54  은평구  ...        3,6   
8031079             0        3241.0        2419.54  은평구  ...        3,6   

        transfer_num  entrance                        gate msi_cnt  hsi_cnt  \
0                  4         9  1, 2, 3, 4, 5, 6, 7, 8, 9,       2        1   
1                  4         9  1, 2, 3, 4, 5, 6, 7, 8, 9,       2        1   
2                  4         9  1, 2, 3, 4, 5, 6, 7, 8, 9,       2        1   
3                  4         9  1, 2, 3, 4, 5, 6, 7, 8, 9,       2        1   
4                  4         9  1, 2, 3, 4, 5, 6, 7, 8, 9,       2        1   
...              ...       ...                         ...     ...      ...   
8031075            2         0                         NaN       2        2   
8031076            2         0                         NaN       2        2   
8031077            2         0                         NaN       2        2   
8031078            2         0                         NaN       2        2   
8031079            2         0                         NaN       2        2   

         uvi_cnt hpi_cnt  mti_cnt   dong  
0              0       0        0    회현동  
1              0       0        0    회현동  
2              0       0        0    회현동  
3              0       0        0    회현동  
4              0       0        0    회현동  
...          ...     ...      ...    ...  
8031075        0       1        1  불광제2동  
8031076        0       1        1  불광제2동  
8031077        0       1        1  불광제2동  
8031078        0       1        1  불광제2동  
8031079        0       1        1  불광제2동  

[8031080 rows x 24 columns]

# '서울특별시' 데이터 추출
- 분석에는 '서울특별시'에 해당하는 자료들만 사용할 것이므로, 지하철 역의 주소가 '서울특별시'에 해당하는 자료들만 추출하도록 하겠다.

In [71]:
subway_final = subway_final[subway_final['address'].str.contains('서울특별시')]
subway_final

date line  station_num station_name boarding  hour  \
0       2018-01-01  1호선          150          서울역       승차     0   
1       2018-01-01  1호선          150          서울역       하차     0   
2       2018-01-01  1호선          150          서울역       승차     5   
3       2018-01-01  1호선          150          서울역       하차     5   
4       2018-01-01  1호선          150          서울역       승차     6   
...            ...  ...          ...          ...      ...   ...   
8031075 2019-12-28  6호선         2615          연신내       하차    21   
8031076 2019-12-28  6호선         2615          연신내       승차    22   
8031077 2019-12-28  6호선         2615          연신내       하차    22   
8031078 2019-12-28  6호선         2615          연신내       승차    23   
8031079 2019-12-28  6호선         2615          연신내       하차    23   

         boarding_num  waiting_area  platform_area   gu  ...   transfer  \
0                  96        8725.0        2080.00   중구  ...  1,4,공항,경의   
1                 147        8725.0        2080.00   중구  ...  1,4,공항,경의   
2                 373        8725.0        2080.00   중구  ...  1,4,공항,경의   
3                 205        8725.0        2080.00   중구  ...  1,4,공항,경의   
4                 318        8725.0        2080.00   중구  ...  1,4,공항,경의   
...               ...           ...            ...  ...  ...        ...   
8031075             0        3241.0        2419.54  은평구  ...        3,6   
8031076             0        3241.0        2419.54  은평구  ...        3,6   
8031077             0        3241.0        2419.54  은평구  ...        3,6   
8031078             0        3241.0        2419.54  은평구  ...        3,6   
8031079             0        3241.0        2419.54  은평구  ...        3,6   

        transfer_num  entrance                        gate msi_cnt  hsi_cnt  \
0                  4         9  1, 2, 3, 4, 5, 6, 7, 8, 9,       2        1   
1                  4         9  1, 2, 3, 4, 5, 6, 7, 8, 9,       2        1   
2                  4         9  1, 2, 3, 4, 5, 6, 7, 8, 9,       2        1   
3                  4         9  1, 2, 3, 4, 5, 6, 7, 8, 9,       2        1   
4                  4         9  1, 2, 3, 4, 5, 6, 7, 8, 9,       2        1   
...              ...       ...                         ...     ...      ...   
8031075            2         0                         NaN       2        2   
8031076            2         0                         NaN       2        2   
8031077            2         0                         NaN       2        2   
8031078            2         0                         NaN       2        2   
8031079            2         0                         NaN       2        2   

         uvi_cnt hpi_cnt  mti_cnt   dong  
0              0       0        0    회현동  
1              0       0        0    회현동  
2              0       0        0    회현동  
3              0       0        0    회현동  
4              0       0        0    회현동  
...          ...     ...      ...    ...  
8031075        0       1        1  불광제2동  
8031076        0       1        1  불광제2동  
8031077        0       1        1  불광제2동  
8031078        0       1        1  불광제2동  
8031079        0       1        1  불광제2동  

[7476280 rows x 24 columns]

# 조인키 생성
- 지하철 데이터 셋을 기준으로 인구 데이터 셋과 기상 데이터 셋을 결합시켜 줄 것이므로, 이를 위한 조인키가 필요하다.
- 따라서 지하철 데이터 셋과 인구 데이터 셋을 결합하기 위한 조인키 1개, 그리고 지하철 데이터 셋과 기상 데이터 셋을 결합하기 위한 조인키 1개를 만들어주도록 하겠다.
    1. 지하철 데이터 셋과 인구 데이터 셋을 결합하기 위한 조인키(pop_join_key)
        - ex) "2018-01-01_회현동"
    2. 지하철 데이터 셋과 기상 데이터 셋을 결합하기 위한 조인키(weather_join_key)
        - ex) "2018-01-01_중구"

In [72]:
# datetime to string
subway_final['date'] = subway_final['date'].astype(str)

# 인구 데이터와 결합할 때 사용할 조인키
subway_final['pop_join_key'] = subway_final['date'] + '_' + subway_final['dong']
# 기상 데이터와 결합할 때 사용할 조인키
subway_final['weather_join_key'] = subway_final['date'] + '_' + subway_final['gu']

<ipython-input-72-575b8d4c54b1>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  subway_final['date'] = subway_final['date'].astype(str)
<ipython-input-72-575b8d4c54b1>:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  subway_final['pop_join_key'] = subway_final['date'] + '_' + subway_final['dong']
<ipython-input-72-575b8d4c54b1>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the docume

In [73]:
subway_final['pop_join_key']

0            2018-01-01_회현동
1            2018-01-01_회현동
2            2018-01-01_회현동
3            2018-01-01_회현동
4            2018-01-01_회현동
                 ...       
8031075    2019-12-28_불광제2동
8031076    2019-12-28_불광제2동
8031077    2019-12-28_불광제2동
8031078    2019-12-28_불광제2동
8031079    2019-12-28_불광제2동
Name: pop_join_key, Length: 7476280, dtype: object

In [74]:
subway_final['weather_join_key']

0           2018-01-01_중구
1           2018-01-01_중구
2           2018-01-01_중구
3           2018-01-01_중구
4           2018-01-01_중구
                ...      
8031075    2019-12-28_은평구
8031076    2019-12-28_은평구
8031077    2019-12-28_은평구
8031078    2019-12-28_은평구
8031079    2019-12-28_은평구
Name: weather_join_key, Length: 7476280, dtype: object

# 불필요한 변수 제거
- station_num, address 변수는 분석에 사용하지 않을 것이므로 제거해주겠다.

In [75]:
subway_final.columns

Index(['date', 'line', 'station_num', 'station_name', 'boarding', 'hour',
       'boarding_num', 'waiting_area', 'platform_area', 'gu', 'address',
       'type', 'length', 'floor', 'transfer', 'transfer_num', 'entrance',
       'gate', 'msi_cnt', 'hsi_cnt', 'uvi_cnt', 'hpi_cnt', 'mti_cnt', 'dong',
       'pop_join_key', 'weather_join_key'],
      dtype='object')

In [76]:
subway_final.drop(['station_num', 'address'], axis = 1, inplace = True)

C:\Users\Brian\anaconda3\lib\site-packages\pandas\core\frame.py:3990: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().drop(


In [77]:
subway_final

date line station_name boarding  hour  boarding_num  \
0        2018-01-01  1호선          서울역       승차     0            96   
1        2018-01-01  1호선          서울역       하차     0           147   
2        2018-01-01  1호선          서울역       승차     5           373   
3        2018-01-01  1호선          서울역       하차     5           205   
4        2018-01-01  1호선          서울역       승차     6           318   
...             ...  ...          ...      ...   ...           ...   
8031075  2019-12-28  6호선          연신내       하차    21             0   
8031076  2019-12-28  6호선          연신내       승차    22             0   
8031077  2019-12-28  6호선          연신내       하차    22             0   
8031078  2019-12-28  6호선          연신내       승차    23             0   
8031079  2019-12-28  6호선          연신내       하차    23             0   

         waiting_area  platform_area   gu type  ...  entrance  \
0              8725.0        2080.00   중구   섬식  ...         9   
1              8725.0        2080.00   중구   섬식  ...         9   
2              8725.0        2080.00   중구   섬식  ...         9   
3              8725.0        2080.00   중구   섬식  ...         9   
4              8725.0        2080.00   중구   섬식  ...         9   
...               ...            ...  ...  ...  ...       ...   
8031075        3241.0        2419.54  은평구  상대식  ...         0   
8031076        3241.0        2419.54  은평구  상대식  ...         0   
8031077        3241.0        2419.54  은평구  상대식  ...         0   
8031078        3241.0        2419.54  은평구  상대식  ...         0   
8031079        3241.0        2419.54  은평구  상대식  ...         0   

                               gate msi_cnt  hsi_cnt  uvi_cnt hpi_cnt  \
0        1, 2, 3, 4, 5, 6, 7, 8, 9,       2        1        0       0   
1        1, 2, 3, 4, 5, 6, 7, 8, 9,       2        1        0       0   
2        1, 2, 3, 4, 5, 6, 7, 8, 9,       2        1        0       0   
3        1, 2, 3, 4, 5, 6, 7, 8, 9,       2        1        0       0   
4        1, 2, 3, 4, 5, 6, 7, 8, 9,       2        1        0       0   
...                             ...     ...      ...      ...     ...   
8031075                         NaN       2        2        0       1   
8031076                         NaN       2        2        0       1   
8031077                         NaN       2        2        0       1   
8031078                         NaN       2        2        0       1   
8031079                         NaN       2        2        0       1   

         mti_cnt   dong      pop_join_key  weather_join_key  
0              0    회현동    2018-01-01_회현동     2018-01-01_중구  
1              0    회현동    2018-01-01_회현동     2018-01-01_중구  
2              0    회현동    2018-01-01_회현동     2018-01-01_중구  
3              0    회현동    2018-01-01_회현동     2018-01-01_중구  
4              0    회현동    2018-01-01_회현동     2018-01-01_중구  
...          ...    ...               ...               ...  
8031075        1  불광제2동  2019-12-28_불광제2동    2019-12-28_은평구  
8031076        1  불광제2동  2019-12-28_불광제2동    2019-12-28_은평구  
8031077        1  불광제2동  2019-12-28_불광제2동    2019-12-28_은평구  
8031078        1  불광제2동  2019-12-28_불광제2동    2019-12-28_은평구  
8031079        1  불광제2동  2019-12-28_불광제2동    2019-12-28_은평구  

[7476280 rows x 24 columns]

In [78]:
subway_final.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7476280 entries, 0 to 8031079
Data columns (total 24 columns):
 #   Column            Dtype  
---  ------            -----  
 0   date              object 
 1   line              object 
 2   station_name      object 
 3   boarding          object 
 4   hour              int64  
 5   boarding_num      int64  
 6   waiting_area      float64
 7   platform_area     float64
 8   gu                object 
 9   type              object 
 10  length            int64  
 11  floor             object 
 12  transfer          object 
 13  transfer_num      int64  
 14  entrance          int64  
 15  gate              object 
 16  msi_cnt           int32  
 17  hsi_cnt           int32  
 18  uvi_cnt           int32  
 19  hpi_cnt           int32  
 20  mti_cnt           int32  
 21  dong              object 
 22  pop_join_key      object 
 23  weather_join_key  object 
dtypes: float64(2), int32(5), int64(5), object(12)
memory usage: 1.3+ GB


# 최종 생성된 데이터 셋을 csv 파일로 내보내기

In [79]:
new_filepath = 'C:/Users/Brian/Desktop/Data Science/Projects/Playdata/dataset/made_by_전처리/train'

In [80]:
subway_final.to_csv('{}/subway_final.csv'.format(new_filepath), index = False)